### PR TITLE
Upgrading cudnn for cuda 11.5, includes zlib for windows

### PR DESCRIPTION
**Pull request by atalman**
Upgrading cudnn for cuda 11.5, includes zlib for windows

### CODE REVIEW

File: windows/internal/cuda_install.bat [hunk 1]
```diff
@@ -182,13 +184,21 @@ if not exist "%SRC_DIR%\temp_build\%CUDA_INSTALL_EXE%" (
...
 if not exist "%SRC_DIR%\temp_build\%CUDNN_INSTALL_ZIP%" (
     curl -k -L "http://s3.amazonaws.com/ossci-windows/%CUDNN_INSTALL_ZIP%" --output "%SRC_DIR%\temp_build\%CUDNN_INSTALL_ZIP%"
     if errorlevel 1 exit /b 1
     set "CUDNN_SETUP_FILE=%SRC_DIR%\temp_build\%CUDNN_INSTALL_ZIP%"
 )
 
+@REM Cuda 8.3+ required zlib to be installed on the path
+echo Installing ZLIB dlls
+curl -k -L "http://s3.amazonaws.com/ossci-windows/zlib123dllx64.zip" --output "%SRC_DIR%\temp_build\zlib123dllx64.zip"
+7z x "%SRC_DIR%\temp_build\zlib123dllx64.zip" -o"%SRC_DIR%\temp_build\zlib"
```
Comment (janeyx99) [inline windows/internal/cuda_install.bat:199]:
```suggestion
7z x "%SRC_DIR%\temp_build\zlib123dllx64.zip" -o "%SRC_DIR%\temp_build\zlib"
```

**Comment by janeyx99** [comment]
what is the test plan for windows updates?

**Comment by atalman** [comment]
> what is the test plan for windows updates?

Will restart yesterdays failed tests: https://app.circleci.com/pipelines/github/pytorch/pytorch/444953/workflows/2f24c542-cbe1-4cd4-a1d1-ab6ca6bac715/jobs/16965039

**Comment by atalman** [comment]
Copied over the dll and rerun the tests on circleCI machine.

**Review by janeyx99 — APPROVED** [review]

**Pull request merged**